packages/skeleton-react/src/lib/components/Segment/Segment.tsx

```diff
@@ -1,113 +1,102 @@
+	indicatorBg = 'preset-filled',
+	indicatorRounded = 'rounded',
+	indicatorClasses = '',
 	// Events
-	onChange,
+	onValueChange = noop,
 	// Children
-	children
+	children,
+	// Zag
+	...zagProps
 }) => {
-	function onSelectionHandler(newValue: string) {
-		value = newValue;
-		if (onChange) onChange(newValue);
-	}
+	const [state, send] = useMachine(
+		radio.machine({
+			id: useId(),
+			onValueChange: (details) => {
+				onValueChange(details.value);
+			}
+		}),
+		{ context: zagProps }
+	);
+	const api = radio.connect(state, send, normalizeProps);
 
 	// Set Context
-	const ctx = {
-		value,
-		name,
-		onSelectionHandler
-	};
+	const ctx = { api };
+
+	// Reactive
+	// FIXME: always returns vertical
+	// const rxOrientation = state.context.orientation === 'vertical' ? 'flex-col' : 'flex-row';
+	const rxOrientation = 'flex-row';
 
 	return (
 		<div
-			role="radiogroup"
-			className={`${base} ${flexDirection} ${background} ${border} ${padding} ${gap} ${rounded} ${width} ${classes}`}
+			{...api.getRootProps()}
+			className={`${base} ${rxOrientation} ${background} ${border} ${padding} ${gap} ${rounded} ${width} ${classes}`}
 			data-testid="segment"
 		>
+			{/* Indicator */}
+			<div {...api.getIndicatorProps()} className={`${indicatorBase} ${indicatorBg} ${indicatorRounded} ${indicatorClasses}`}></div>
+			{/* Items */}
 			<SegmentContext.Provider value={ctx}>{children}</SegmentContext.Provider>
 		</div>
 	);
 };
 
 const SegmentItem: React.FC<SegmentItemsProps> = ({
-	id,
 	value,
-	title,
-	disabled = false,
 	// Root
-	base = 'btn',
-	active = 'preset-filled',
-	hover = 'hover:preset-tonal',
+	base = 'btn cursor-pointer z-[1]',
 	classes = '',
 	// Label
 	labelBase = 'pointer-events-none',
+	labelActiveText = 'preset-filled',
 	labelClasses = '',
-	// Events
-	onClick,
 	// Children
 	children
 }) => {
 	// Get Context
-	const ctx = useContext<SegmentContextState>(SegmentContext);
+	const ctx = useContext(SegmentContext);
 
 	// Reactive
-	const [selected, setSelected] = useState(value === ctx.value);
-	const [rxActive, setRxActive] = useState(selected ? active : hover);
-
-	useEffect(() => {
-		setSelected(value === ctx.value);
-	}, [value, ctx.value]);
-
-	useEffect(() => {
-		setRxActive(selected ? active : hover);
-	}, [selected, active, hover]);
-
-	function onClickHandler() {
-		ctx.onSelectionHandler(value);
-		onClick?.(value);
-	}
+	const rxActive = ctx.api.value === value;
+	const rxActiveText = rxActive ? labelActiveText : '';
 
 	return (
-		<button
-			role="radio"
-			aria-checked={selected}
-			onClick={onClickHandler}
-			type="button"
-			className={`${base} ${rxActive} ${classes}`}
-			title={title}
-			disabled={disabled}
-			data-testid="segment-item"
-		>
-			{/* Input */}
-			{selected ? <input type="hidden" name={ctx.name} id={id} value={value} /> : null}
+		<label key={value} {...ctx.api.getItemProps({ value: value })} className={`${base} ${classes}`} data-testid="segment-item">
```

Comment:
`key` is only required for looping in react, I assume you copied this from the Zag example. You can remove the `key` property here.